File: linked_list.py
class Node():
    def __init__(self,data):
        self.data= data
        self.link = None

def insert(start, data):
    node = start
    while node.link:
        node = node.link
    avail = Node(data)
    avail.link = None
    node.link = avail

def display(start):
    node = start
    while node:
        print(node.data)
        node = node.link
    print

def reverse(start):
    node = start
    prev = None
    while node:
        new_start = node.link
        node.link = prev
        prev = node
        node = new_start
    return prev

start = Node(3)
insert(start, 5)
insert(start, 17)
insert(start, 43)
insert(start, 10)
display(start)
start=reverse(start)
display(start)

File: linked_list.cpp
#include<iostream>
using namespace std;
class list{
      struct node{
         string data;
         node* link;
         }*start;
      public:
             list();
             void insert(string);
             void del();
             void display();
      };

list::list(){
     start=NULL;
     }
     
void list::insert(string str){
     node *avail, *p;
     avail=new node;
     avail->data=str;
     avail->link=start;
     start=avail;
     }
     
void list::del(){
     node *p;        
     p=start;
     start=start->link;
     delete p;
     }

void list::display(){
     node*p;
     for(p=start;p->link!=NULL;p=p->link)
        cout<<p->data<<" -> ";
     cout<<p->data<<endl;
     }
     
int main()
{
    list list1;
    list1.insert("Hi");
    list1.display();
    list1.insert("awesome");
    list1.display();
    list1.insert("list");
    list1.display();
    list1.insert("nice");
    list1.display();
    list1.del();
    list1.display();
    list1.del();
    list1.display();
    list1.del();
    list1.display();
}
    

